eth/api_tracer.go

```diff
@@ -719,6 +719,30 @@ func (api *PrivateDebugAPI) TraceTransaction(ctx context.Context, hash common.Ha
 	return api.traceTx(ctx, msg, vmctx, statedb, config)
 }
 
+// Call lets you trace a given eth_call. returns the structured logs created during the execution of EVM
+// if the given transaction was added on top of the provided block and returns them as a JSON object.
+// You can provide -2 as a block number to trace on top of the pending block.
+func (api *PrivateDebugAPI) TraceCall(ctx context.Context, args ethapi.CallArgs, blockNrOrHash rpc.BlockNumberOrHash, config *TraceConfig) (interface{}, error) {
+	// First try to retrieve the state
+	statedb, header, err := api.eth.APIBackend.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+	if err != nil && blockNrOrHash.BlockHash != nil {
```

Comment:
If we can't construct the state directly(e.g. state is pruned), the error is not nil.
But actually no matter the block hash or block number is specified, we can all convert them to "canonical" block hash. Here we only recompute the state for `blockhash` specified RPC calls.

What's more, if the call doesn't specify the hash and number, an error will also be returned. In this case, we should return the error directly.

Reply:
Sounds good, I try to retrieve the block now by number + hash and if I didn't find them I return the original error.